### PR TITLE
buf fix & refactoring

### PR DESCRIPTION
**Pull request by sfzeng**
Fix a bug about transiting and migrating archived object.
Fix a bug - [https://github.com/opensds/multi-cloud/issues/134](url)

### CODE REVIEW

File: datamover/pkg/drivers/https/prepare.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+			}
+			return &loca, nil
+		}
+	default:
+		{
+			logger.Printf("unsupport type:%s.\n", conn.Type)
+			return nil, errors.New("unsupport type")
+		}
+	}
+}
```
Comment (leonwanghui) [inline datamover/pkg/drivers/https/prepare.go]: Need to return some values here.

File: datamover/pkg/drivers/https/prepare.go [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+func getObjs(ctx context.Context, in *pb.RunJobRequest, defaultSrcLoca *LocationInfo, offset, limit int32) ([]*osdss3.Object, error) {
+	switch in.SourceConn.Type {
+	case flowtype.STOR_TYPE_OPENSDS:
+		return getOsdsS3Objs(ctx, in, offset, limit)
+	default:
+		{
```
Comment (leonwanghui) [inline datamover/pkg/drivers/https/prepare.go]: No need to add `{}` here.

File: datamover/pkg/drivers/https/prepare.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+				case "security":
+					loca.Security = cfg[i].Value
+				default:
+					logger.Printf("unknow key[%s] for connector.\n", cfg[i].Key)
+				}
+			}
+			return &loca, nil
+		}
+	default:
+		{
```
Comment (leonwanghui) [inline datamover/pkg/drivers/https/prepare.go]: No need to add `{}` here.

File: datamover/pkg/drivers/https/prepare.go [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+			reqbk := osdss3.Bucket{Name: virtBkname}
+			rspbk, err := s3client.GetBucket(ctx, &reqbk)
+			if err != nil {
+				logger.Printf("get bucket[%s] information failed when refresh connector location, err:%v\n", virtBkname, err)
+				return nil, errors.New("get bucket information failed")
+			}
+			return getOsdsLocation(ctx, virtBkname, rspbk.Backend)
+		}
+	case flowtype.STOR_TYPE_AWS_S3, flowtype.STOR_TYPE_HW_OBS, flowtype.STOR_TYPE_HW_FUSIONSTORAGE, flowtype.STOR_TYPE_AZURE_BLOB, flowtype.STOR_TYPE_CEPH_S3, flowtype.STOR_TYPE_GCP_S3, flowtype.STOR_TYPE_IBM_COS:
+		{
```
Comment (leonwanghui) [inline datamover/pkg/drivers/https/prepare.go]: No need to add `{}` here.

File: datamover/pkg/drivers/https/prepare.go [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+			VirBucket: virtBkname, Access: bk.Access, Security: bk.Security, BakendName: backendName}
+		logger.Printf("Refresh backend[name:%s,id:%s] successfully.\n", backendName, bk.Id.String())
+		return loca, nil
+	}
+}
+
+func getConnLocation(ctx context.Context, conn *pb.Connector) (*LocationInfo, error) {
+	switch conn.Type {
+	case flowtype.STOR_TYPE_OPENSDS:
+		{
```
Comment (leonwanghui) [inline datamover/pkg/drivers/https/prepare.go]: No need to add `{}` here.

File: datamover/pkg/drivers/https/prepare.go [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+	logger.Printf("count objects of bucket[%s]: count=%d,size=%d\n", in.SourceConn.BucketName, count, size)
+	return rsp.Count, rsp.Size, nil
+}
+
+func countObjs(ctx context.Context, in *pb.RunJobRequest) (count, size int64, err error) {
+	switch in.SourceConn.Type {
+	case flowtype.STOR_TYPE_OPENSDS:
+		return countOsdsS3Objs(ctx, in)
+	default:
+		{
```
Comment (leonwanghui) [inline datamover/pkg/drivers/https/prepare.go]: No need to add `{}` here.

Comment (sfzeng) [reply datamover/pkg/drivers/https/prepare.go]: fixed

File: datamover/pkg/drivers/https/prepare.go (hunk 3 above)
Comment (sfzeng) [reply datamover/pkg/drivers/https/prepare.go]: fixed

File: datamover/pkg/drivers/https/prepare.go (hunk 2 above)
Comment (sfzeng) [reply datamover/pkg/drivers/https/prepare.go]: fixed

File: datamover/pkg/drivers/https/prepare.go (hunk 4 above)
Comment (sfzeng) [reply datamover/pkg/drivers/https/prepare.go]: fixed

File: datamover/pkg/drivers/https/prepare.go (hunk 5 above)
Comment (sfzeng) [reply datamover/pkg/drivers/https/prepare.go]: fixed

File: datamover/pkg/drivers/https/prepare.go (hunk 1 above)
Comment (sfzeng) [reply datamover/pkg/drivers/https/prepare.go]: fixed

**Pull request merged**